src/S3/PostObjectV4.php

```diff
@@ -56,7 +56,7 @@ public function __construct(
         $credentials   = $this->client->getCredentials()->wait();
         $securityToken = $credentials->getSecurityToken();
 
-        if (null !== $securityToken) {
+        if (!empty($securityToken)) {
```

Comment:
I am not sure if using `if ($securityToken = $credentials->getSecurityToken()) {` will check for `isEmpty($securityToken)`. Doesn't it only check for truth/false values?